crates/wit-component/src/encoding.rs

```diff
@@ -493,6 +505,10 @@ impl<'a> EncodingState<'a> {
         let ty = encoder.ty;
         // Don't encode empty instance types since they're not
         // meaningful to the runtime of the component anyway.
+        //
+        // TODO: Is this correct? What if another imported interface needs to
+        // alias a type exported by this interface but can't because we skipped
+        // encoding the import?
```

Comment:
In theory this is handled by a "liveness pass" somewhere where if only an interface's types are needed those are included but the functions are pruned. That being said there's been bugs around this in the past.